System/Posix/DynamicLinker.hsc

```diff
@@ -52,14 +52,12 @@ import System.Posix.DynamicLinker.Prim
 #include "HsUnix.h"
 
 import Control.Exception        ( bracket )
-import Control.Monad    ( liftM )
 import Foreign
 import System.Posix.Internals ( withFilePath )
 
 dlopen :: FilePath -> [RTLDFlags] -> IO DL
-dlopen path flags = do
-  withFilePath path $ \ p -> do
-    liftM DLHandle $ throwDLErrorIf "dlopen" (== nullPtr) $ c_dlopen p (packRTLDFlags flags)
+dlopen path flags = withFilePath path $ \ p -> DLHandle <$> do
+  throwDLErrorIf "dlopen" (== nullPtr) $ c_dlopen p (packRTLDFlags flags)
```

Comment:
```suggestion
dlopen path flags = withFilePath path $ \p -> DLHandle <$>
  throwDLErrorIf "dlopen" (== nullPtr) (c_dlopen p (packRTLDFlags flags))
```

Reply:
I'm Curious why you don't like "do" blocks as values?

Reply:
I don't like do-blocks with a single statement inside. You are basically using it instead of a pair of brackets, which is better supported by text editors and eyes. 

This is a minor stylistic thing, I do not insist.

Reply:
I don't like do-blocks with a single statement inside. You are basically using it instead of a pair of brackets, which is better supported by text editors and eyes. 

This is a minor stylistic thing, I do not insist.

Reply:
Done in upcoming push.